packages/contracts-bedrock/scripts/go-ffi/utils.go

```diff
@@ -127,3 +128,40 @@ func (n *proofList) Put(key []byte, value []byte) error {
 func (n *proofList) Delete(key []byte) error {
 	panic("not supported")
 }
+
```

Comment:
What value does ffi based differential fuzzing provide vs a roundtrip encode/decode in regular solidity?

Reply:
thought of using solidity's `encode`/`decode` but wanted something different from what's implemented inside the library's logic to differentially test, so opted to differentially test against abi encoding in go (as is done with the `CrossDomainMessenger` encoding)

Reply:
Just curious the exact benefits of differential testing here - for the cross domain messages, the differential fuzzing existed because we used Go to manipulate storage slots during the migration to bedrock, so we wanted to be sure that Go and solidity generated the same thing

Reply:
the main benefit I see is testing that the library encodes values properly against a different implementation of encoding, catching a fault in the encoding logic of the library. what do you think? checking against by using abi.encode in the test feels not enough, because it's the same logic underpinning the library itself (the test will always pass)